packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -285,10 +292,34 @@ export class MonacoWorkspace implements lang.Workspace {
                 model.pushStackElement();
                 totalFiles += 1;
                 totalEdits += editOperations.length;
-            }
+            });
+            const ariaSummary = this.getAriaSummary(totalEdits, totalFiles);
+            return { ariaSummary };
+        } catch (e) {
+            return { ariaSummary: `Error applying workspace edits: ${e.toString()}` };
         }
-        const ariaSummary = this.getAriaSummary(totalEdits, totalFiles);
-        return { ariaSummary };
+    }
+
+    protected async openEditors(editorEdits: EditsByEditor[], options?: EditorOpenerOptions) {
+        return Promise.all(
+            editorEdits.map(async editorEdit => {
+                const editorWidget = await this.editorManager.open(new URI(editorEdit.uri), options);
```

Comment:
I do like this change, as in theory it should make it faster to open the editors.

But it seems like in practice it is still kinda slow, I don't know why...

Reply:
I sequentialized that again and now it works. 

Opening editors is a big task, e.g. requires fetching the content form the backend. Nothing we can do against that in this PR.

Reply:
> Opening editors is a big task, e.g. requires fetching the content form the backend. Nothing we can do against that in this PR.

I agree, but I still wonder why we do open all the editors in the first place?

It looks really bad when the refactoring targets 10+ files, maybe just open the list of files impacted in some widget on the side? Something similar to the `Find All References` from code:

![image](https://user-images.githubusercontent.com/14182238/54449667-0d0f9d80-4725-11e9-9cbe-116123b58b21.png)

But you are right that this is out of the scope of this PR, I just wanted to have your opinion.

Reply:
It's the default VSCode / LSP behavior as applying WorkspaceEdits requires an editor.